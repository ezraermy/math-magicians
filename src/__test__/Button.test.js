// Import dependencies
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from '../components/button';

describe('Button component', () => {
  it('renders a button with the correct label and class', () => {
    const mockOnClick = jest.fn();
    const { container } = render(
      <Button label="Click me!" onClick={mockOnClick} className="btn" />,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('calls the onClick function when the button is clicked', () => {
    const mockOnClick = jest.fn();
    render(<Button label="Click me!" onClick={mockOnClick} className="btn" />);
    const button = screen.getByRole('button', { name: /Click me!/i });
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
