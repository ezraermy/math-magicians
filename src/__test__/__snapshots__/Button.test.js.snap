// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a button with the correct label and class 1`] = `
<input
  class="btn"
  type="button"
  value="Click me!"
/>
`;
